elastic/beats
# [metricbeat] - Allow metricsets to report their status via v2 protocol

## Proposed commit message

- Add StatusReporter for letting individual metricsets to report their statuses.
- Add `NonFatalError` support for `system/process` summary
    - See https://github.com/elastic/elastic-agent-system-metrics/pull/172 for non-fatal errors.

## Checklist




## Disruptive User Impact



## Author's Checklist



## How to test this PR locally

- How to test with agent:
  - Build `agentbeat` locally from this branch.
  - Download/Build `elastic-agent`.
  - Put the `agentbeat` binary in `elastic-agent/data/elastic-agent-*/components/`
  - Install the agent in `unprivileged` mode and observe the health.
       - You should see `DEGRADED` state in a couple of minutes.
       - You should also see `non-fatal error fetching PID metrics` and `permission denied` errors.

## Related issues


- Closes https://github.com/elastic/beats/issues/39736
- Closes https://github.com/elastic/beats/issues/39737

## Use cases



## Screenshots

![Screenshot 2024-07-18 at 10 43 38 PM](https://github.com/user-attachments/assets/2c696ccc-cd78-40db-95c5-3964ed78c755)
